### DISCUSSION TITLE
WireGuard only works on Android but not on Manjaro Linux

### DISCUSSION THREAD

**Discussion by BaumGuard** [pivpn/pivpn#1640] [Support]
## In raising this issue, I confirm the following:

- [X] I have read the [documentation](https://docs.pivpn.io)
- [X] Is this a support request? please consider copying the content of this document and using it to open a [Discussion](https://github.com/pivpn/pivpn/discussions/new?category=support)
- [X] I have read and understood the [PiVPN General Guidelines](https://github.com/pivpn/pivpn/#general-guidelines)
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Has your install failed? 
```
no
```

### Describe the issue
I have installed WireGuard on my Raspberry Pi 4 using PiVPN and added a few clients on it, e.g. I'm using WireGuard on my Android phone and on my notebook with Manjaro. I can connect to the VPN from my mobile phone without any problems and can access devices in my LAN and the internet. However, the connection doesn't work on Manjaro although I have tried a lot of things to fix it. I can import the client's config file using the network manager. When I enable the VPN, it says that it's connected but I can't access the internet then. It doesn't work either when I'm trying to connect to it using `wg-quick`. Some time ago it worked when using the `kernel 4.19.257-1` but on all kernel versions starting with a 5 I have no chance to get it up and running.
When I try to connect to the VPN from my Manjaro system and execute `pivpn -c` on the Raspberry Pi, it says **(not yet)** for **Last seen**.

**Expected behavior**
My Manjaro system should connect to the VPN and be able to access the LAN and the internet, just as my Android phone does.

### Have you searched for similar issues and solutions?
    yes, but none of them worked for me.

**Additional context**
This is the client's config file:
```
[Interface]
PrivateKey = ___Private_Key___
Address = 10.6.0.6/24
DNS = 8.8.8.8, 8.8.4.4

[Peer]
PublicKey = ___Public_Key___
PresharedKey =___Preshared_Key___
Endpoint = ___FritzBox-Domain___:51820
AllowedIPs = 0.0.0.0/0, ::0/0
```

### Have you taken any steps towards solving your issue?
```
I've read many issues on GitHub and forum entries on the internet but none of them worked for me.
```

#### What type of hardware are you running PiVPN at?

```
Raspberry Pi 4
```
#### Output of `uname -a`
```
  Raspberry Pi: Linux rpi4 5.10.103-v7+ #1529 SMP Tue Mar 8 12:21:37 GMT 2022 armv7l GNU/Linux
  Manjaro: Linux manjaro 5.19.7-1-MANJARO #1 SMP PREEMPT_DYNAMIC Mon Sep 5 10:16:57 UTC 2022 x86_64 GNU/Linux
```

#### Output of `cat /etc/os-release`
```
**Raspberry Pi:**
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

**Manjaro:**
NAME="Manjaro Linux"
ID=manjaro
ID_LIKE=arch
BUILD_ID=rolling
PRETTY_NAME="Manjaro Linux"
ANSI_COLOR="32;1;24;144;200"
HOME_URL="https://manjaro.org/"
DOCUMENTATION_URL="https://wiki.manjaro.org/"
SUPPORT_URL="https://manjaro.org/"
BUG_REPORT_URL="https://bugs.manjaro.org/"
LOGO=manjarolinux
```

### Console output of      `pivpn add` or `pivpn add nopass`
```
  Enter a name for the client:
```

**Comment by coolapso** [comment]
Moved to discussions. 

Please make sure that you are not connecting with your laptop when inside the same network as the VPN server. 
Try connecting the laptop to the Phone Mobile hotspot and test the connection over the Mobile network. 

Connecting to the VPN while in the same network as the server does not work.

**Comment by BaumGuard** [comment]
Sorry for the late reply. In the last two weeks it was working flawlessly when connecting to the WireGuard server from outside my LAN. A few days ago, I've installed updates on my system and now it doesn't work anymore again, although I'm still outside of the VPN's LAN.

**Reply by coolapso** [reply]
check with `pivpn -d`, sometimes it detects some common issues and suggests to fix them. otherwise just paste the output here. now it can be a completely different issue.

**Reply by BaumGuard** [reply]
This is the output of  `pivpn -d`:

```
::::            PiVPN debug              ::::
=============================================
::::            Latest commit            ::::
Branch: master
Commit: ab1fe203992dc0ef586644efb12488b876ed9555
Author: 4s3ti
Date: Sat Oct 8 02:39:53 2022 +0200
Summary: fix(ufw): set missing openvpn ipv6 variables
=============================================
::::        Installation settings        ::::
PLAT=Raspbian
OSCN=buster
USING_UFW=0
IPv4dev=eth0
IPv4addr=192.168.178.30/24
IPv4gw=192.168.178.1
install_user=pi
install_home=/home/pi
VPN=wireguard
pivpnPORT=51820
pivpnDNS1=8.8.8.8
pivpnDNS2=8.8.4.4
pivpnHOST=REDACTED
INPUT_CHAIN_EDITED=0
FORWARD_CHAIN_EDITED=0
pivpnPROTO=udp
pivpnDEV=wg0
pivpnNET=10.6.0.0
subnetClass=24
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=()
=============================================
::::  Server configuration shown below   ::::
[Interface]
PrivateKey = server_priv
Address = 10.6.0.1/24
ListenPort = 51820
### begin h-jonas-laptop ###
[Peer]
PublicKey = h-jonas-laptop_pub
PresharedKey = h-jonas-laptop_psk
AllowedIPs = 10.6.0.2/32
### end h-jonas-laptop ###
### begin h-jonas-handy ###
[Peer]
PublicKey = h-jonas-handy_pub
PresharedKey = h-jonas-handy_psk
AllowedIPs = 10.6.0.3/32
### end h-jonas-handy ###
=============================================
::::  Client configuration shown below   ::::
[Interface]
PrivateKey = h-jonas-laptop_priv
Address = 10.6.0.2/24
DNS = 8.8.8.8, 8.8.4.4

[Peer]
PublicKey = server_pub
PresharedKey = h-jonas-laptop_psk
Endpoint = REDACTED:51820
AllowedIPs = 0.0.0.0/0, ::0/0
=============================================
::::    Recursive list of files in       ::::
::::    /etc/wireguard shown below       ::::
/etc/wireguard:
configs
keys
wg0.conf

/etc/wireguard/configs:
clients.txt
h-jonas-handy.conf
h-jonas-laptop.conf

/etc/wireguard/keys:
h-jonas-handy_priv
h-jonas-handy_psk
h-jonas-handy_pub
h-jonas-laptop_priv
h-jonas-laptop_psk
h-jonas-laptop_pub
server_priv
server_pub
=============================================
::::            Self check               ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] WireGuard is running
:: [OK] WireGuard is enabled
(it will automatically start on reboot)
:: [OK] WireGuard is listening on port 51820/udp
=============================================
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://docs.pivpn.io/faq
=============================================
:::: WARNING: This script should have automatically masked sensitive       ::::
:::: information, however, still make sure that PrivateKey, PublicKey      ::::
:::: and PresharedKey are masked before reporting an issue. An example key ::::
:::: that you should NOT see in this log looks like this:                  ::::
:::: YIAoJVsdIeyvXfGGDDadHh6AxsMRymZTnnzZoAb9cxRe                          ::::
=============================================
::::            Debug complete           ::::
:::
::: Debug output completed above.
::: Copy saved to /tmp/debug.log
:::
```

As before, Linux indicates a connection to the VPN but I have no internet access and when I run `pivpn -c` it shows an older date for `Last  seen`.

**Comment by BaumGuard** [comment]
Somehow, it's working again now, at least when I connect to the VPN using wg-quick. It doesn't work when starting it from the network manager. But the web traffic seems to bypass the VPN: When I monitor the network traffic in Wireshark and open a website in the browser, I can find the domain names as DNS requests in the data, although it should all be encrypted.

**Comment by DrRek** [comment]
Hey there, I'm having the same issue, no one else has ever faced this?

**Reply by BaumGuard** [reply]
I found a workaround some months ago. Somehow the DNS resolution doesn't work, so I just replaced the URL by the IP address of the WireGuard server in the client's config file and it works.

**Reply by DrRek** [reply]
Thanks, then I guess we have a different problem
